ED/src/dynamics/structural_growth.f90

```diff
@@ -1121,7 +1152,17 @@ subroutine plant_structural_allocation(ipft,hite,dbh,lat,phen_status,elongf,bdea
       !------------------------------------------------------------------------------------!
       if (f_bseeds < r_tol_trunc) f_bseeds = 0.0
       if (f_growth < r_tol_trunc) f_growth = 0.0
-      f_bstorage = 1.0 - f_growth - f_bseeds
+
+      ! plants will save bstorage_reserve as mobile carbon supply for potential reflushing
+      ! we take the maximum of the residual of f_bseeds and f_growth and bstorage_reserve 
+      ! / bstorage to maintain compatibility with previous versions (bstorage_reserve = 0.)
+      f_bstorage = max(1.0 - f_bseeds - f_growth, bstorage_reserve / bstorage)
+
+      ! we need to modify f_bseeds and f_growth accordingly if f_bseeds+f_growth is non-zero
+      if (f_bseeds + f_growth > 0.0) then
+          f_bseeds = f_bseeds / (f_bseeds + f_growth) * (1.0 - f_bstorage)
```

Comment:
@xiangtaoxu I think we should do the following instead:
```
f_bgi    = 1.0 / (f_bseeds + f_growth)
f_bseeds = f_bseeds * f_bgi * (1.0 - f_bstorage)
f_growth = f_growth * f_bgi * (1.0 - f_bstorage)
```

Otherwise f_bseeds is already modified by the time you update f_growth and the sum may not be 1.0.

Reply:
Good catch! Thanks!